### PR TITLE
Add zlib

### PR DESCRIPTION
**Pull request by crazyhappygame**

### CODE REVIEW

File: src/external/zlib/gzguts.h [hunk 1]
```diff
@@ -207,3 +207,6 @@ char ZLIB_INTERNAL *gz_strwinerror OF((DWORD error));
 unsigned ZLIB_INTERNAL gz_intmax OF((void));
 #  define GT_OFF(x) (sizeof(int) == sizeof(z_off64_t) && (x) > gz_intmax())
 #endif
+
+
+#include <unistd.h>
```
Comment (crazyhappygame) [inline src/external/zlib/gzguts.h]: To fix iOS build

File: src/engine/engine.cpp [hunk 2]
```diff
@@ -1,19 +1,31 @@
...
 {
     std::ostringstream o;
-    o << "engine info\n";
+    o << "External libs info\n";
     o << "Box2D version: " << b2_version.major << "." << b2_version.minor
       << "." << b2_version.revision << "\n";
 
     dtNodeQueue q(10);
     o << "Recast dtNodeQueue: " << q.empty() << "\n";
+    o << "zlib version: " << zlibVersion() << "\n";
```
Comment (crazyhappygame) [inline src/engine/engine.cpp:19]: @drelaptop I propose to add similar call for each library

File: src/external/zlib/CMakeLists.txt.org [hunk 3]
```diff
@@ -0,0 +1,249 @@
+cmake_minimum_required(VERSION 2.4.4)
```
Comment (crazyhappygame) [inline src/external/zlib/CMakeLists.txt.org:1]: File was renamed 
mv CMakeLists.txt CMakeLists.txt.org

File: src/external/zlib/gzguts.h [hunk 4]
```diff
@@ -207,3 +207,11 @@ char ZLIB_INTERNAL *gz_strwinerror OF((DWORD error));
...
 #  define GT_OFF(x) (sizeof(int) == sizeof(z_off64_t) && (x) > gz_intmax())
 #endif
+
+
+#ifdef __APPLE__
+#include "TargetConditionals.h"
+#if TARGET_OS_IPHONE
+#include <unistd.h>
+#endif
+#endif
```
Comment (crazyhappygame) [inline src/external/zlib/gzguts.h:217]: To fix iOS build

**Comment by crazyhappygame** [comment]
@drelaptop could you merge it?

File: src/engine/engine.cpp (hunk 2 above)
Comment (drelaptop) [reply src/engine/engine.cpp:19]: agree, this is a good test case to confirm library works

**Comment by drelaptop** [comment]
with pleasure. sorry for a long response

**Pull request merged**